### DISCUSSION TITLE
How to switch between animations (not speed, but states like Idle, Running)

### DISCUSSION THREAD

**Discussion by stevenliebregt** [jcornaz/benimator#48] [Q&A]
Hi, I'm struggling with figuring out how I could switch between 2 different animation sets. I have a texture atlas for idle animations and 1 for running animations. How do I switch between those two?

**Comment by jcornaz** [comment]
You can replace the animation handle with the handle of the animation you would like to run. You may check out [this example](https://github.com/jcornaz/benimator/blob/main/examples/change_animation.rs)

**Reply by stevenliebregt** [reply]
Yeah but that just changes the speed of the animation right? I want to switch the actual animation images.

**Reply by jcornaz** [reply]
It changes the *animation* being played. In the case of the example, it happens to be a similar animation with only a different speed. But the animation can be completely different: different frames, different looping, different frame order, etc.

**Reply by stevenliebregt** [reply]
Maybe I'm understanding it wrong, but the textures are not part of the SpriteSheetAnimation or Res<Animations>, so currently I have a player_idle.png with 4 frames, and a player_run.png with 4 frames. As I'm understanding now I have 2 options:

1. Keep the separate files and switch the SpriteSheetBundle instead of the animation
2. Create a file that contains both idle and run frames, then switch the range, say idle is frame 0..=3 and run is frame 4..=7 then I'd just change those values

**Reply by jcornaz** [reply]
If the sprites are in a different texture you also need to swap the texture handle.

Personally, I put every animation of a single character/object in the same sprite sheet. Therefore I only need to change the animation. But that's not required, you can also change the texture handle when you change the animation handle.
